### PR TITLE
Fix: #4695 Reference CommonJS child bundles with a require call

### PR DESCRIPTION
**Pull request by jfrconley**
# ↪️ Pull Request

This pull request adds support to JSRuntime for referencing CommonJS child bundles with a require call. No handling of this was present, so the current behavior was to use a URL reference.

Resolves: #4695

## 💻 Examples

There isn't an easy way to see this in action without a custom bundler. I created a simple one (https://github.com/jfrconley/parcel-bundler-splitable) that allows listing files you want to split the bundle on. The integration test relies on a custom bundler with a hardcoded path to split on. 

## 🚨 Test instructions

Testing is covered by added integration test: commonjs-bundle-require

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by jfrconley** [comment]
@mischnic Is there a timeline for when this can be merged? Keeping a custom runtime plugin up to date is causing some issues for us.

**Comment by mischnic** [comment]
I'll take a look.
Could you slim down the bundler plugin you added in the test so that it only includes the bare minimum to make the test work? For example the `optimize` step likely doesn't do anything here

**Comment by jfrconley** [comment]
I removed optimize and deduplicate. Additionally, I've resolved merge conflicts and pulled in upstream changes

**Comment by jfrconley** [comment]
@mischnic Is there any other cleanup I need to do before this can be merged?

File: packages/core/integration-tests/test/integration/commonjs-bundle-require/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "engines": {
+    "node": "8"
+  },
+  "targets": {
+    "main": {
+      "engines": {
+        "node": "8"
+      }
+    }
+  }
```
Comment (mischnic) [inline packages/core/integration-tests/test/integration/commonjs-bundle-require/package.json:17]:
```suggestion
  }
```

File: packages/core/integration-tests/test/integration/commonjs-bundle-require/node_modules/parcel-bundler-splitable/package.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "parcel-bundler-splitable",
+  "version": "1.2.3"
```
Comment (mischnic) [inline packages/core/integration-tests/test/integration/commonjs-bundle-require/node_modules/parcel-bundler-splitable/package.json:3]:
```suggestion
  "version": "1.2.3",
  "engines": {
    "parcel": "^2.0.0-alpha.1"
  }
```

**Comment by mischnic** [comment]
I would have suggested support ESM as well while we're at it, but that doesn't work yet for other reasons...

**Pull request merged**